### PR TITLE
chore(event): resolve strictNullCheck errors in event-decorator

### PR DESCRIPTION
**Pull request by rwaskiewicz**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Unit tests (`npm test`) were run locally and passed
- [x] E2E Tests (`npm run test.karma.prod`) were run locally and passed
- [x] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

`event-decorator.ts` contains a handful of strict null checks errors
GitHub Issue Number: N/A

## What is the new behavior?

this commit updates the function signatures in `event-decorator.ts` to resolve a few `strictNullChecks` violations in the file. it also adds JSDoc to the file

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

Type checker continues to pass

### CODE REVIEW

**Review by tanner-reits — APPROVED** [review]

File: src/compiler/transformers/decorators-to-static/event-decorator.ts [hunk 1]
```diff
@@ -27,12 +27,22 @@ export const eventDecoratorsToStatic = (
...
 
+/**
+ * Parse a single instance of Stencil's `@Event()` decorator and generate metadata for the class member that is
+ * decorated
+ * @param diagnostics a list of diagnostics used as a part of the parsing process. Any parse errors/warnings shall be
+ * added to this collection
+ * @param typeChecker an instance of the TypeScript type checker, used to generate information about the `@Event()` and
+ * its surrounding context in the AST
+ * @param prop the property on the Stencil component class that is decorated with `@Event()`
+ * @returns generated metadata for the class member decorated by `@Event()`
```
Comment (alicewriteswrongs) [inline src/compiler/transformers/decorators-to-static/event-decorator.ts:38]: should we indicate the possible `null` return type here in `@returns` ?

**Review by alicewriteswrongs — CHANGES_REQUESTED** [review]
just had one question, looks good otherwise!

File: src/compiler/transformers/decorators-to-static/event-decorator.ts (hunk 1 above)
Comment (rwaskiewicz) [reply src/compiler/transformers/decorators-to-static/event-decorator.ts]: 👍 done in 5c296cfb90aca387ab6ccf9fbf043be3fd444f73

**Review by alicewriteswrongs — APPROVED** [review]
lgtm!

**Pull request merged**